### ISSUE TITLE
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with: status = StatusCode.INVALID_ARGUMENT

### ISSUE THREAD

**Issue by wangyunxiaa** [tensorflow/serving#1202]
I have a tf model which contains a LookupTable, and I used
table_init_op = tf.tables_initializer(name="init_all_tables") config.sess.run(table_init_op)
to initialize it, and freezed the graph by
output_graph_def = convert_variables_to_constants(config.sess, config.sess.graph_def, output_node_names=['init_all_tables', 'Inference/final_output/output', 'Inference/cur_state'])
finally I save the model by builder.save(), it is ok until i run my tf server, however, when i do inference, I got an errorgrpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with: status = StatusCode.INVALID_ARGUMENT details = "NodeDef mentions attr 'Truncate' not in Op<name=Cast; signature=x:SrcT -> y:DstT; attr=SrcT:type; attr=DstT:type>; NodeDef: Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/Cast = Cast[DstT=DT_INT32, SrcT=DT_BOOL, Truncate=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/GreaterEqual). (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.). [[Node: Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/Cast = Cast[DstT=DT_INT32, SrcT=DT_BOOL, Truncate=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/GreaterEqual)]]" debug_error_string = "{"created":"@1543839204.502783269","description":"Error received from peer","file":"src/core/lib/surface/call.cc","file_line":1083,"grpc_message":"NodeDef mentions attr 'Truncate' not in Op<name=Cast; signature=x:SrcT -> y:DstT; attr=SrcT:type; attr=DstT:type>; NodeDef: Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/Cast = Cast[DstT=DT_INT32, SrcT=DT_BOOL, Truncate=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/GreaterEqual). (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).\n\t [[Node: Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/Cast = Cast[DstT=DT_INT32, SrcT=DT_BOOL, Truncate=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Inference/affine_layers2/out_2/affine_transform/dense/Tensordot/GreaterEqual)]]","grpc_status":3}"
does the table can freeze to the graph,what should i do to freeze it?
thank you very much for your reply~

**Comment by harshini-gadige** [comment]
@wangyunxiaa  Could you answer the following which help us to look into this issue.

1) If you are using Tensorflow Serving, what is the version ?
2) What is the version of Tensorflow(not serving) installed ?
3) Can you try the same on Saved model CLI and see if this error is reproduced there.

**Comment by wangyunxiaa** [comment]
Hi，
I have tried "Saved model CLI" and got the correct  result. 
my server tfs version is 1.5
client tf version is 1.4.1, python: 2.7
and I am confused that I used "table = tf.contrib.lookup.index_table_from_tensor(mapping=mapping_strings, num_oov_buckets=1, default_value=133294, name='table')"
Do this variables can be freezed into the graph?

**Comment by wangyunxiaa** [comment]
In addition,
The tensorflow version of the generated model is 1.10.0 
the tensorflow serving version is 1.5.0 
the tensorflow version of do inference on server is 1.4.1
does the different version cause the problem?

**Comment by wangyunxiaa** [comment]
@harshini-gadige @lilao @ewilderj @kchodorow 
Hello,everyone
I have a test and I am confused about the result,which is that I added the output names when i freeze the graph by 
`for table_init_op in tf.get_collection(tf.GraphKeys.TABLE_INITIALIZERS):
            OUTPUT_NAMES.append(table_init_op.name)
        output_graph_def = convert_variables_to_constants(config.sess,
                                                         config.sess.graph_def,
                                                         output_node_names=OUTPUT_NAMES)`
and saved the model by `with tf.gfile.FastGFile('graph_simple.pb', mode='wb') as f:
           f.write(output_graph_def.SerializeToString())`  and  builder.save().
when i do inference on my own computer, I added a line code of `sess.run(loaded_graph.get_operation_by_name('table/hash_table/table_init'))` 
in addition 'table/hash_table/table_init' is the op name of table init operation. by this way, I can do correctly result.
but when I run the saved model on server, the saved model was added `main_op=sess.run(graph.get_operation_by_name(table_init_ops[0].name))` , I always get the error of `status = StatusCode.FAILED_PRECONDITION
	details = "Table not initialized.`
I have checked that the model has the map_string table, however, It should be initialized manually.
I want to know how to init the table on tensorflow serving.
than you very much for your reply~
